src/zenml/metadata_stores/mysql_metadata_store.py

```diff
@@ -37,11 +42,93 @@ def get_tfx_metadata_config(
         metadata_store_pb2.ConnectionConfig,
         metadata_store_pb2.MetadataStoreClientConfig,
     ]:
-        """Return tfx metadata config for mysql metadata store."""
-        return metadata.mysql_metadata_connection_config(
+        """Return tfx metadata config for MySQL metadata store."""
+        config = MySQLDatabaseConfig(
             host=self.host,
             port=self.port,
             database=self.database,
-            username=self.username,
-            password=self.password,
         )
+
+        secret = self._get_mysql_secret()
+
+        # Set the user
+        if self.user:
+            if secret and secret.user:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a username `{self.user}` and `{secret.user}`. Please "
+                    f"make sure that you only use one."
+                )
+            else:
+                config.user = self.user
+        else:
+            if secret and secret.user:
+                config.user = self.user
+            else:
+                raise RuntimeError(
+                    "Your metadata store does not have a username. Please "
+                    "provide it either by defining it upon registration or "
+                    "through a MySQL secret."
+                )
+
+        # Set the password
+        if self.password:
+            if secret and secret.password:
+                raise RuntimeError(
+                    f"Both the metadata store {self.name} and the secret "
+                    f"{self.secret} within your secrets manager define "
+                    f"a password. Please make sure that you only use one."
+                )
+            else:
+                config.password = self.password
+        else:
+            if secret and secret.user:
+                config.password = self.password
+
+        # Set the SSL configuration if there is one
+        if secret:
+            ssl_options = {}
+            # Handle the files
+            for key in ["ssl_key", "ssl_ca", "ssl_cert"]:
+                content = getattr(secret, key)
+                filepath = os.path.join(os.getcwd(), f"{key}.pem")
```

Comment:
If I were concerned about security, I would be a bit concerned about using the current folder to store these files, especially because there is no cleanup done afterwards. I have some ideas about how to handle this:

* (preferable) use a designated fixed location unique for each metadata store. I would recommend using the global config directory as a root folder, with the component UUID reflected in one of the sub-directories, same as we do for local components. This approach will also work when running in a Kubeflow container.

* use a temporary folder (mktemp) and make sure to remove it after. It might be difficult to asses when the files are no longer needed. After instantiating the `ConnectionConfig` object ? After instantiating the `metadata_store.MetadataStore` object ? Or maybe the files need to be present throughout the entire lifetime of the connection ?

Reply:
Good catch! You are absolutely right and I wanted to change this as well. I understand the appeal of both options but I also share the concern that you have with the temp folder. That's why I implemented the first approach.